Sources/BraintreePayPal/BTPayPalApprovalURLParser.swift

```diff
@@ -45,8 +45,8 @@ struct BTPayPalApprovalURLParser {
         return nil
     }
 
-    init?(body: BTJSON, linkType: String?) {
-        if linkType == "universal", let payPalAppRedirectURL = body["agreementSetup"]["paypalAppApprovalUrl"].asURL() {
+    init?(body: BTJSON) {
+        if let payPalAppRedirectURL = body["agreementSetup"]["paypalAppApprovalUrl"].asURL() {
```

Comment:
For EditFI work, we are not using payPalAppApprovalUrl because app switching is not in scope yet for that work stream. But if the gateway returns this value as well as approvalURL, using this Parser we would get payPalAppRedirectURL instead of the webURL. For right now, I just forced the web Url fetching  by passing in linkType.

Reply:
The specs on Confluence show: Response - agreementSetup {tokenId, paypalAppApprovalUrl, approvalUrl}
as response for edit FI url endpoint. Maybe the doc is not updated.

Reply:
Okay, then removing the link type will not impact that flow and we should not need to pass in a link type, it will always fail this first check for a `payPalAppApprovalUrl` and move on to the next block which includes the web URL.

Reply:
I've posted the question on the collaboration channel so we can clarify with them.

Reply:
Fwiw we also have to opt in to the app switch flow for the Gateway to return that property, so I would expect it only exists to future proof this feature. We need to pass them everything here to say the merchant opted into app switch which we won't be doing for Edit FI currently: https://github.com/braintree/braintree_ios/blob/38297e4a697a1b89a1addab9c42c7af8553b8a57/Sources/BraintreePayPal/BTPayPalVaultRequest.swift#L56-L64

The gateway only returns a `payPalAppApprovalUrl` if those properties are passed AND the merchant is eligible on the gateway side.

Reply:
Great! thank you for the clarification. I do want to make sure that this is true for editFI endpoint as well since there are no details given in the specs. I want to clarify that this will not be returned for generate_edit_fi_url endpoint if these parameters are not passed.